kong/cmd/utils/prefix_handler.lua

```diff
@@ -609,6 +651,12 @@ local function prepare_prefix(kong_config, nginx_custom_template_path, skip_writ
     return true
   end
 
+  -- check lmdb directory
+  local ok, err = pre_create_lmdb(kong_config)
```

Comment:
Instead of "creating the file as root" (and then having to do all those `chmod`s), would it be possible for the privileged process to "wait and yield", locking until some other worker creates the files with the right permissions?

```
if process.type() == "privileged agent" then
  wait_for_someone_else_to_create_lmdb_file()
end
```

Reply:
Perhaps we can not do it.

Because lua-resty-lmdb create the files in C land init_worker phase, it is earlier than ngx_lua.

Reply:
This is not a good idea, it introduces artificial race condition and is generally bad design IMO.

Reply:
Alright then, thanks for considering